### ISSUE TITLE
Optimize the Current ConcurrentHashMap usage

### ISSUE THREAD

**Issue by He-Pin** [apache/incubator-pekko#588]
The `ConcurrentHashMap` on Java 8 have some performance issue around the `computeIfAbsent`.
see https://bugs.openjdk.org/browse/JDK-8161372 , which fixed in Java 9.

A quick fix should be do a quick `get` first.

eg: https://github.com/apache/dubbo/blob/2a5b437d58ba60cd51d39a040b6d39c8697f9d50/dubbo-common/src/main/java/org/apache/dubbo/common/utils/ConcurrentHashMapUtils.java#L28

**Comment by pjfanning** [comment]
I don't think we should do this. For me, we need proof that this is a big issue before doing anything. That Dubbo change doesn't even look correct to me. The hack to split the computeIfAbsent call over multiple calls changes the way the code works. We should look at how other teams work around this issue and not just copy the solution from one team.

**Comment by He-Pin** [comment]
Caffeine does the same trick, and which was fixed in Java 9,

**Comment by mdedetrich** [comment]
Im okay with this as long as serious testing is done both to see we don't cause issues wrt concurrency and we confirm its faster

**Comment by He-Pin** [comment]
This will bring some performance optimization for where read happens most the time. One want to contributes this can do it with jmh benchmark.
